defaults/main.yml

```
@@ -1,13 +1,13 @@
 ---
 jitsi_meet_server_name: "meet.example.com"
 jitsi_meet_videobridge_loglevel: "ERROR"
-jitsi_meet_videobridge_secret: "CHANGEME"
+jitsi_meet_videobridge_secret: "{{ lookup('password', 'credentials/' + ansible_hostname + '/jitsi_meet_videobridge_secret length=16') }}"
```

What about
```suggestion
jitsi_meet_videobridge_secret: "{{ vault_jitsi_meet_videobridge_secret }}"
```
and not providing a default at all? This fails if the user has not provided a secret via the vault and does not permit deploying a server with an unsafe default.

---

I do that sometimes, but in general I would consider that an unnecessary hurdle.

The only change I'd make, on second thought, would be to generate the secrets to `chars=hexdigits length=32`. I could change that when others feel there is merrit in merging anyway!

---

I see the advantage here to generate a secure default secret. I will discuss that with my collegues. Thanks for the PR and helping to improve this role ans especially the security and the risk of having unsecure secrets!

---

I'm fine with having @mdik solution. It's the straight forward and easy usable [ansible way](https://docs.ansible.com/ansible/latest/plugins/lookup/password.html) to do it.

@shoetten s solution is also valid and from a security aspect also a good one, but will lead to more effort in testing. Also I'm quite okay with random generated passwords > 12 characters.

---

I like chars=hexdigit better, because depending on the markup language config files are written in, and how they interpret quotation marks (or not), passwords with characters that may be interpreted syntactically, are something I try to avoid. and I make them longer to compensate.

I have no idea how to resolve the conflict, technically, as I am not very well versed in git :-s (and it says "Only those with write access to this repository can merge pull requests.")

---

We merged #10 meanwhile, that adds more secrets we need to change in that way. So you need to rebase your branch on the latest master and resolve the conflicts and add the password lookups for the new secrets.

---

Are `jitsi_meet_videobridge_password` and `jitsi_meet_videobridge_muc_nickname` exposed to users (in general), i.e. do length and charset matter?

---

If we automate this step with generated passwords, imho it doesn't matter if the password is very 32 characters long. The `jitsi_meet_videobridge_muc_nickname` variable is in defaults a UUID. I don't know if you can't solve this with the password lookup